x/blob/keeper/params.go

```diff
@@ -21,12 +22,22 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
 
 // MinSquareSize returns the MinSquareSize param
 func (k Keeper) MinSquareSize(ctx sdk.Context) (res uint32) {
+	// use the default size for the first block so that we return a value on the
+	// first block in PrepareProposal
+	if ctx.BlockHeader().Height < 1 {
```

Comment:
I'm pretty sure the first block is at height 1 not 0. Also we should be using something like `InitialHeight` as it's possible with Tendermint chains to start at any height. 

Most importantly, I'm still not sure if this logic is necessary. At least I haven't seen this in other SDK modules.

Reply:
I'm not sure either. However, with this logic removed, prepare proposal (where the params are fetched for the first time) blocks and using the cli to query the params panics with an error

Reply:
chain logs
```
5:47PM INF Timed out dur=10000 height=1 module=consensus round=0 step=3
5:47PM INF Ensure peers module=pex numDialing=0 numInPeers=0 numOutPeers=0 numToDial=10
5:47PM INF No addresses to dial. Falling back to seeds module=pex
5:48PM INF Ensure peers module=pex numDialing=0 numInPeers=0 numOutPeers=0 numToDial=10
5:48PM INF No addresses to dial. Falling back to seeds module=pex
```

using `celestia-appd query blob params` results in
```
Error: rpc error: code = Unknown desc = UnmarshalJSON cannot decode empty bytes: panic
```

Reply:
Interesting. Maybe you're right. I'd have thought the genesis params would be set at `InitChain` which comes before `PrepareProposal` but maybe I'm wrong.

What you could also do is add a log for when the params first get set in the Keeper and we can see if that comes before the first block gets proposed. Also wouldn't this panic you're seeing also happen in `PrepareProposal`?